PasswordVerifier/src/com/company/PasswordVerifier.java

```diff
@@ -0,0 +1,81 @@
+public class PasswordVerifier {
+
+    public static void main(String[] args) {
+        String pass = "abdsdscdEF";
+        System.out.println("Password is : "+pass);
+        try {
+            System.out.println("Response is : "+verify(pass));
+        } catch (InvalidPasswordException e) {
+            System.out.println("Response is : "+e.getMessage());
+        }
+    }
+
+    public static String verify(String pass) throws InvalidPasswordException {
+        int count=0;
+        boolean isupper=false;
+        boolean isdigit = false;
+        if(pass==null || pass.trim().isEmpty() || pass.length()<8) {
+            throw new InvalidPasswordException("Password must contains atleast 8 characters");
+        }else {
+            count++;
+        }
+        if(!pass.matches("(.*)[a-z](.*){1,}")) {
```

Comment:
Regular expressions are generally quite hard to read. You could extract this check into a method called `containsLowercase`. Then this would read:
```java
if (!containsLowercase(password) {
```

This is much easier for someone unfamiliar with the code to read.

Reply:
You could also move the whole if statement into a method. It might look something like this:
```java
public void validateContainsLowercase(String password) {
    if (!pass.matches("(.*)[a-z](.*){1,}")) {
        throw new InvalidPasswordException("Password must contains atleast one lowercase letter");
    }
}
```

Then you could rewrite this as:
```java
validateContainsLowercase(password);
count++
```